src/api/utils/request_api.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Request is a a custom Network Layer.  Its main benefits of using it over fetch include
+ * a longer chain of promises which allows us to write error handling code all in one place.
+ * Request will give json back or some standard, meaningful error so that the errors and parsing
+ * don't have to be handled higher up in the call stack.
+ */
+
+class Request {
+  static headers() {
+    return {
+      'Content-Type': 'application/json'
+    }
+  }
+
+  static get(url) {
+    return this.fetchWrapper(url, null, 'GET')
+  }
+  static post(url, params) {
+    return this.fetchWrapper(url, params, 'POST')
+  }
+
+  static isJSON(response) {
+    // returns true if response is of type JSON
+    const contentType = response.headers.get('content-type');
+
+    return contentType && contentType.includes('application/json');
+  }
+
+  /**
+   * @param {Response} response
+   * @return {Promise} 
+   */
+  static parseResponseBody(response) {
+    if (this.isJSON(response)) {
+      return response.json();
+    }
+
+    return response.text();
+  }
+
+  /**
+   * @param {String} url
+   * @param {Object?} params
+   * @param {String} verb
+   * @return {Promise} 
+   */
+  static fetchWrapper(url, params, verb) {
+
+    // for logging/debugging
+    console.log(`${verb} ${url}`, params);
+
+    // baseURL is URL without params
+    // useful if settings/filters are in URL
+    const baseURL = url.split('?')[0];
+
+    // options are the complicated object of options that are passed in a fetch response
+    // another benefit of abstracting this portion away
+    let options = Object.assign({ method: verb }, params ? { body: JSON.stringify(params) } : null);
+    options.headers = Request.headers();
+
+    let fetchResponse, fetchStatusCode, errorResponse;
+
+    return fetch(url, options).then(response => {
```

Comment:
Personally I prefer async await but this works. How easy is it to change?

Reply:
I thought about this too when i first started writing but I chose this syntax because I personally thought it is easier to read if you're new to promises or es6.  But can update if you feel strongly about.

Reply:
it's nbd 🍨

Reply:
already ded